spec.html

```diff
@@ -5570,6 +5571,23 @@ <h1>Strict Equality Comparison</h1>
   <emu-clause id="sec-operations-on-objects">
     <h1>Operations on Objects</h1>
 
+    <emu-clause id="sec-allocatebasicobject" aoid="AllocateBasicObject">
+      <h1>AllocateBasicObject ( _internalSlotsList_ )</h1>
+      <p>The abstract operation AllocateBasicObject is the source of all ECMAScript objects, including both ordinary objects and exotic objects. It factors out common steps used in creating all objects, and centralizes object creation.</p>
+
+      <emu-alg>
+        1. Assert: _internalSlotsList_ is a List of internal slot names.
+        1. Let _obj_ be a newly created object with an internal slot for each name in _internalSlotsList_.
```

Comment:
if the list is empty, are these first two steps still coherent?

Reply:
An empty List is a List, if that's what you mean. Sometimes we say "a possibly empty List" if we want to emphasize it.

Reply:
Does "for each name in the list" also work when there are none?

I can buy that both of these lines still work in this case, I'm just making sure :-)

Reply:
> Does "for each name in the list" also work when there are none?

Well, the spec doesn't *say* it works -- it doesn't talk about *any* of its looping constructs -- but there are definitely other cases of 'For each' where the thing being iterated over can be empty. (My guess is that this is true of the majority of 'For each' steps.)

Reply:
I think the intention is clear enough here that we don't need to fully spell out in-prose looping constructs.